api/v1alpha1/oidc_types.go

```diff
@@ -106,7 +106,17 @@ type OIDC struct {
 }
 
 // OIDCProvider defines the OIDC Provider configuration.
+// +kubebuilder:validation:XValidation:message="BackendRefs must be used, backendRef is not supported.",rule="!has(self.backendRef)"
 type OIDCProvider struct {
+	// BackendCluster defines the backend cluster to be used for the OIDC Provider.
```

Comment:
`BackendCluster` is an internal term, imo we shouldnt expose it to the user, `cluster` is a new term the user will need to understand which may confuse them with a k8s cluster